src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,147 @@
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class RegistrationServiceTest {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+    private static final int EDGE_SHIFT = 1;
+    private RegistrationService registrationService;
+    private User testingUser;
+
+    @BeforeEach
+    void init() {
+        testingUser = new User("testingLogin", "testingPassword", 25);
+        registrationService = new RegistrationServiceImpl();
+        Storage.people.addAll(List.of(
+                new User("login1", "password1", 18),
+                new User("login2", "password2", 20),
+                new User("login3", "password3", 21),
+                new User("login4", "password4", 22),
+                new User("login5", "password5", 23)
+        ));
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_validData_ok() {
+        User returnedUser = registrationService.register(testingUser);
+        User registeredUser = Storage.people.stream()
+                .filter(user -> user.getLogin().equals(testingUser.getLogin()))
+                .findFirst()
+                .orElse(null);
+        assertEquals(testingUser, returnedUser);
+        assertNotNull(registeredUser);
+        assertTrue(Storage.people.contains(testingUser));
+        assertEquals(testingUser, registeredUser);
+
+        assertEquals(testingUser.getId(), registeredUser.getId());
+        assertEquals(testingUser.getLogin(), registeredUser.getLogin());
+        assertEquals(testingUser.getPassword(), registeredUser.getPassword());
+        assertEquals(testingUser.getAge(), registeredUser.getAge());
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        testingUser.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(testingUser));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        testingUser.setPassword(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(testingUser));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        testingUser.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(testingUser));
```

Comment:
Add tests for a user exactly 18 years old and over 18 years old to cover all possible edge cases regarding age.